working/1426-extension-types/feature-specification-implements-rep.md

```diff
@@ -0,0 +1,227 @@
+# Inline Classes can Implement the Representation Type
+
+Author: eernst@google.com
+
+Status: Draft.
+
+This is an addendum to the [inline class feature specification][],
+proposing an additional feature: An inline class `V` may have its
+representation type `R` as a superinterface. This causes `V` to be a
+subtype of `R`, and it enables invocations of members of `R` on a receiver
+of type `V`.
+
+[inline class feature specification]: https://github.com/dart-lang/language/blob/main/accepted/future-releases/inline-classes/feature-specification.md
+
+## Change Log
+
+## Summary
+
+This is a proposal to add a new feature to inline classes: An inline class `V`
+may include a superinterface in its `implements` clause which is a
+supertype `R0` of the ultimate representation type `R` of `V` (this allows
```

Comment:
Should "supertype" here be "superinterface"?

That is, can an inline class on a `List<int>` choose to implement `Iterable<num>`?

Can it implement both `List<num>` and `Iterable<int>`, and therefore presumably also `Iterable<num>`, if we assume the shared members end up having a most specific signature?

Reply:
I don't see why we should prevent an inline class with representation type `List<int>` to have `implements Iterable<num>` and then be a subtype of `Iterable<num>` and have members from `Iterable` accordingly.

I haven't considered allowing any declaration with superinterfaces to have two or more distinct generic instantiations in their superinterface graph. In other words, `Iterable<num>` and `Iterable<int>` in the same superinterface graph is an error.

We could certainly create an expression of type `V<num>` whose representation object is a `List<int>`, which means that there is nothing at run-time which can occur when this is allowed which couldn't already occur. So it would be kind of an empty safety measure, just prohibiting something which can occur anyway.